plugins/woocommerce/templates/auth/form-grant-access.php

```diff
@@ -42,6 +42,13 @@
 	<?php endforeach; ?>
 </ul>
 
+<p>
+	<?php
+	/* Translators: %s Domain name from the callback URL. */
+	printf( esc_html__( 'Credentials will be shared with the website %s.', 'woocommerce' ), '<strong>' . esc_html( wp_parse_url( $callback_url, PHP_URL_HOST ) ) . '</strong>' );
```

Comment:
A few small/optional things, just wanted to briefly discuss them:

1. Should we drop the use of the word 'website'? It implies that accessing the domain should lead to a website of some kind, which may not always be the case.
2. I guess this is implicit, but I'm wondering about structuring the sentence something like: *"If you approve this request, credentials will be shared with &lt;DOMAIN&gt;. You should not proceed if you have any concerns about the validity of that domain."* Overkill?

If we prefer to go ahead without these changes, totally fine: it works well, is already a nice enhancement and I'm hitting approve.